### DISCUSSION TITLE
Autoload preset based on system color scheme

### DISCUSSION THREAD

**Discussion by MrFacundo** [luisbocanegra/plasma-panel-colorizer#60] [General]
Is it possible to autoload the preset based on the current KDE color scheme?
Or set keyboard shortcuts to switch the color preset?

**Comment by luisbocanegra** [comment]
Hi,
Currently colors are reloaded with the color scheme only if System colors are used:

![Screenshot_20240707_124636](https://github.com/luisbocanegra/plasma-panel-colorizer/assets/15076387/de69bf83-0cb6-490d-ab64-dbaff9780f36)

Still, there are other elements like shadows that only allow fixed colors and I pan to add that in https://github.com/luisbocanegra/plasma-panel-colorizer/issues/40

Switching between presets (Preset auto-loading tab) based on the color scheme or a keyboard shortcut is not currently supported.

Color scheme detection could be implemented by running custom commands to get the available color schemes `plasma-apply-colorscheme -l` and other to get the current one, probably reading `~/.config/kdeglobals`. Or using native C++ code (cleaner but probably harder with my current knowledge)

The shortcuts may be a bit harder to implement as I don't think is currently supported to register a shortcut using only QML so C++ will probably be needed, if implemented should work like the custom blur regions (WIP) it will be optional in case the user installs without compiling.

A very hacky and manual alternative to both could be adding a key for a preset name that will be loaded when the widget is loaded, then with an external script, change that key value in `~/.config/plasma-org.kde.plasma.desktop-appletsrc` and restart plasmashell.

Or maybe register a dbus service, either python ([example](https://github.com/luisbocanegra/plasma-cursor-eyes)) or C++ that runs on every instance and allows doing things like listing and applying presets

**Comment by MrFacundo** [comment]
Thanks for the reply. On my case, selecting the right system colors and the setting up the autoload option did the job. Now the colors change according to the system color theme.
